spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -431,9 +436,12 @@ else if (theListener instanceof BatchAcknowledgingMessageListener) {
 			else {
 				throw new IllegalArgumentException("Listener must be one of 'MessageListener', "
 						+ "'BatchMessageListener', 'AcknowledgingMessageListener', "
-						+ "'BatchAcknowledgingMessageListener', not " + theListener.getClass().getName());
+						+ "'BatchAcknowledgingMessageListener', not " + this.theListener.getClass().getName());
 			}
 			Assert.state(!this.isBatchListener || !this.isRecordAck, "Cannot use AckMode.RECORD with a batch listener");
+			if (this.autoCommit && this.theListener instanceof ConsumerSeekAware) {
+				((ConsumerSeekAware) this.theListener).registerSeekCallback(this);
```

Comment:
Yes, we don't have a test case; will add one - good catch - I had a similar problem with the other case - I called from the ctor instead of the run method in the `ListenerInvoker`. See the second commit where I moved that code.